xlators/features/shard/src/shard.c

```diff
@@ -4831,6 +4839,30 @@ shard_open(call_frame_t *frame, xlator_t *this, loc_t *loc, int32_t flags,
     return 0;
 }
 
+int
+shard_opendir(call_frame_t *frame, xlator_t *this, loc_t *loc, fd_t *fd,
+              dict_t *xdata)
+{
+    int ret = 0;
+
+    xdata = xdata ? dict_ref(xdata) : dict_new();
```

Comment:
We need to do error checking for `dict_new()`.

Reply:
done

Reply:
done